express-notes-api/index.js

```diff
@@ -0,0 +1,99 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+app.use(express.json());
+
+function ObjtoArr(obj) {
+  const arr = [];
+  for (const id in obj) {
+    arr.push(obj[id]);
+  }
+  return arr;
+}
+
+app.get('/api/notes', function (req, res) {
+  fs.readFile('data.json', 'utf8', (err, data) => {
+    if (err) throw err;
+    const obj = JSON.parse(data);
+    const notesArr = ObjtoArr(obj.notes);
+    res.status(200).send(notesArr);
+  });
+});
+
+app.get('/api/notes/:id', function (req, res) {
+  const idNum = parseInt(req.params.id);
+  if (isNaN(idNum) || idNum < 1) {
+    return res.status(400).send({ error: 'id must be a positive integer' });
+  }
```

Comment:
These routes should always send a `400 Bad Request` if `id` isn't a positive integer:

---

![image](https://user-images.githubusercontent.com/16143771/143174424-f4651e2f-8ac1-4c8b-a2cf-331c65c23b9c.png)

---

![image](https://user-images.githubusercontent.com/16143771/143174436-934bec0f-a003-4749-9a4e-196df4856b36.png)